hadml/train.py

```diff
@@ -76,7 +76,7 @@ def train(cfg: DictConfig, logger=None) -> Tuple[dict, dict]:
     datamodule: LightningDataModule = hydra.utils.instantiate(cfg.datamodule)
 
     log.info(f"Instantiating model <{cfg.model._target_}>")
-    model: LightningModule = hydra.utils.instantiate(cfg.model, datamodule=datamodule)
+    model: LightningModule = hydra.utils.instantiate(cfg.model)
```

Comment:
Instead of modifying this, can we add datamodule as an argument in the relevant module? Similarly as done here: https://github.com/hep-lbdl/hadml/blob/0b701a76da47e5a59522374eee59b4166280338c/hadml/models/cgan/cond_particle_gan.py#L64

It's not (for now) needed for event training, but it will be when we introduce particle types and we need to use the gumbel softmax distribution. This fix would break `CondParticleGANModule`